server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -283,9 +284,12 @@ static List<JoinPair> convertImplicitJoinConditionsToJoinPairs(List<JoinPair> ex
         while (queryIterator.hasNext()) {
             Map.Entry<Set<RelationName>, Symbol> queryEntry = queryIterator.next();
             Set<RelationName> relations = queryEntry.getKey();
+            Symbol implicitJoinCondition = queryEntry.getValue();
+            if (implicitJoinCondition instanceof io.crate.expression.symbol.Function func && CaseFunction.NAME.equals(func.name())) {
+                continue;
+            }
```

Comment:
Why is case special here?

Reply:
Also, shouldn't we check with a visitor? can it be that the `CASE` is not on top level of the condition? e.g: 
```
WHERE (t1.c0 > CASE 'q' WHEN t1.c1 THEN 5 ELSE 10 END );
```

Reply:
Thank you both, I assumed that `convertImplicitJoinConditionsToJoinPairs` did not account for `case`. I think the bug should also be fixed by https://github.com/crate/crate/pull/15677(keeping the `case` expr in join condition) which I will proceed first then get back to this.

Reply:
I still wonder if `convertImplicitJoinConditionsToJoinPairs` can consider only EqOperators to be implicit join conditions.

Reply:
What I'm asking for is some kind of user scenario that doesn't work, some explanation of why it doesn't work, and some explanation why this is the correct fix.

We shouldn't end up adding a new if statement here whenever we encounter a new statement variant that uses another expression.

Reply:
Sorry, update the description with more details.

Reply:
Thanks for pointing out, as you said all other functions go thorugh the same logic and merged to join conditions. If `case` breaks, it should be handled else where.